perfkitbenchmarker/providers/ibmcloud/ibmcloud_disk.py

```diff
@@ -0,0 +1,206 @@
+
+    volcmd = ibm.IbmAPICommand(self)
+    volcmd.flags.update({
+      'name': self.name,
+      'instanceid': vm.vmid,
+      'volume': self.vol_id,
+      'delete': True
+      })
+
+    # wait till volume is PROVISIONED before attach
+    status = None
+    endtime = time.time() + _MAX_DISK_ATTACH_SECONDS
+    while time.time() < endtime:
+      status = json.loads(volcmd.ShowVolume())['status']
+      logging.info('Checking volume status: %s', status)
+      if status == ibm.States.AVAILABLE:
+        logging.info('Volume is available')
+        break
+      time.sleep(2)
+
+    if status != ibm.States.AVAILABLE:
+      logging.error('Failed to create a volume')
+      raise errors.Error('IBMCLOUD ERROR: failed to provision a volume.')
+
+    resp = json.loads(volcmd.InstanceCreateVolume())
+    logging.info('Attached volume, volcmd.flags %s', volcmd.flags)
+    volume = resp['id']
+    logging.info('VDISK ID: %s, vdiskname: %s, instanceid: %s', volume, self.name, vm.vmid)
+    self.attached_vdisk_URI = volume
+    self.attached_vm = vm
+    self._WaitForVolumeAttachment(vm)
+    self._GetDeviceFromVDisk(vm)
+
+  @vm_util.Retry(poll_interval=1, max_retries=-1, timeout=_MAX_DISK_ATTACH_SECONDS, log_errors=False,
+                 retryable_exceptions=errors.Resource.RetryableCreationError)
+  def _WaitForVolumeAttachment(self, vm):
+    """Waits and checks until the volume status is attached """
+    if self.vol_id is None:
+      return
+    volcmd = ibm.IbmAPICommand(self)
+    volcmd.flags.update({
+      'instanceid': self.attached_vm.vmid,
+      'volume': self.vol_id
+      })
+    logging.info('Checking volume on instance %s, volume: %s', vm.vmid, self.vol_id)
+    status = None
+    endtime = time.time() + _MAX_DISK_ATTACH_SECONDS
+    while time.time() < endtime:
+      resp = volcmd.InstanceShowVolume()
+      if resp:
+        resp = json.loads(resp)
+        status = resp.get('status', 'unknown')
+        logging.info('Checking instance volume status: %s', status)
+        if status == ibm.States.ATTACHED:
+          logging.info('Remote volume %s has been attached to %s', self.name, vm.name)
+          break
+      time.sleep(2)
+    if status != ibm.States.ATTACHED:
+      logging.error('Failed to attach the volume')
+      raise errors.Error('IBMCLOUD ERROR: failed to attach a volume.')
+
+  def _GetDeviceFromVDisk(self, vm):
+    """Gets device path for the volume """
+    cmd = 'fdisk -l'
+    endtime = time.time() + _MAX_FIND_DEVICE_SECONDS
+    self.device_path = None
+    while time.time() < endtime:
+      stdout, _ = self.attached_vm.RemoteCommand(cmd, should_log=True)
+      # parse for lines that contain disk size in bytes
+      disks = re.findall(r'\Disk (\S+): .* (\d+) bytes,', stdout)
+      for device_path, disk_size in disks:
+        logging.info('disk_path: %s, disk_size: %s', device_path, disk_size)
+        if int(disk_size) >= self.disk_size * 1000000000:
```

Comment:
I would use scientific notation here:
  self.disk_size * 1e9
as easier to see that you mean 1 billion.

Reply:
Nice.